### PR TITLE
Handheld Mass Scanner

### PR DESCRIPTION
**Pull request by TheShuEd**
## About the PR
Handheld mass scanner from Frontier fork. Has 64 radius of view (unlike the standard 256 on the console)
Powered by Power Cell
## Why / Balance
useful for not getting too lost in space with a jetpack. 

## Media
image

https://github.com/space-wizards/space-station-14/assets/96445749/4b8a3db8-b785-4588-847a-880ae40e38e9

https://github.com/space-wizards/space-station-14/assets/96445749/f11f204a-854e-41e2-a806-8a1b7ad5864c

- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

**Changelog**
:cl:
- add: Added a new T2 industrial technology "Space Scanning"! It includes a mass scanner console board, a Borg GPS module, and a new handheld mass scanner.
- add: GPS borg module now have handheld mass scanner.

### CODE REVIEW

**Comment by 0x6273** [comment]
I don't think this should be roundstart since it pretty much makes the GPS obsolete.

**Comment by Nimfar11** [comment]
A handheld mass scanner does not show coordinates, which will not save you at long distances. But yes, in most situations it will replace GPS. Maybe as T1 or even T2 technology.

**Comment by Ghagliiarghii** [comment]
Agree that roundstart is too strong, would make a T2 Industrial unlock for maybe 5000 points? Other than that it looks cool :+1:

**Comment by TheShuEd** [comment]
image
moved to T2

**Comment by FungiFellow** [comment]
Please consider putting one into the GPS Borg Module in this PR, or I can just make a PR for it after this comes out.

Here's the Code to make one that has unlimited power and can't have the cell removed toss this with your other versions
**handheld_mass_scanner.yml**
```
id: HandHeldMassScannerBorg
parent: HandHeldMassScanner
  suffix: Borg
   components:
- type: ItemSlots
   slots:
     cell_slot:
         name: power-cell-slot-component-slot-name-default
         startingItem: PowerCellMicroreactor
         disableEject: true
         swap: false
```

Could also make an unpowered variant like this
```
- type: entity
  id: HandHeldMassScannerUnpowered
  parent: BaseHandHeldMassScanner
  suffix: Handheld, Unpowered
```

Then just plug it into **borg_modules.yml** under id: BorgModuleGPS in Items

**Comment by FungiFellow** [comment]
My bad

**Comment by TheShuEd** [comment]
image

**Comment by FungiFellow** [comment]
Might want to include a screenshot for a Borg using it.

**Comment by TheShuEd** [comment]
https://github.com/space-wizards/space-station-14/assets/96445749/bbbe4608-66bf-4a1f-81a9-1d396e97a6ad

File: Content.Server/Shuttles/Systems/RadarConsoleSystem.cs [hunk 1]
```diff
@@ -30,6 +32,13 @@ protected override void UpdateState(EntityUid uid, RadarConsoleComponent compone
...
+        //handheld test
+        if (HasComp<PowerCellDrawComponent>(uid))
+        {
+            coordinates = new EntityCoordinates(uid, Vector2.Zero);
+            angle = Angle.Zero + MathHelper.DegreesToRadians(180);
+        }
+
         // Use ourself I guess.
         if (TryComp<IntrinsicUIComponent>(uid, out var intrinsic))
         {
```
Comment (metalgearsloth) [inline Content.Server/Shuttles/Systems/RadarConsoleSystem.cs:44]: 1. Did you ask the frontier coder as a courtesy.
2. This shouldn't be using a separate UI path to intrinsic UIs considering they're doing the same thing.

**Pull request merged**